cg/apps/lims/api.py

```diff
@@ -68,6 +63,13 @@ def samples_in_pools(self, pool_name, projectname):
         """Fetch all samples from a pool"""
         return self.get_samples(udf={"pool name": str(pool_name)}, projectname=projectname)
 
+    def get_source(self, lims_id: str) -> str | None:
+        """Return the source from LIMS for a given sample ID.
+        Return 'None' if no source information is set or
+        if sample is not found or cannot be fetched from LIMS."""
+        if lims_sample := self.sample(lims_id=lims_id):
+            return lims_sample.get("source")
```

Comment:
No need of an `if` as we discussed:
```suggestion
        lims_sample: dict[str, Any] = self.sample(lims_id=lims_id)
        return lims_sample.get("source")
```

Reply:
you're right. Banana brain...